### PR TITLE
Add support for stablehlo from tt-mlir project.

### PR DESCRIPTION
**Pull request by uazizTT**
Add support for stablehlo from tt-mlir project as an embedded project. 

Stablehlo is disabled by default and can be enabled by turning on TTMLIR_ENABLE_STABLEHLO flag.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -39,6 +44,7 @@ set(TTMLIR_TOOLCHAIN_DIR $ENV{TTMLIR_TOOLCHAIN_DIR})
 set(TTMLIR_SOURCE_DIR ${PROJECT_SOURCE_DIR})
 set(TTMLIR_BINARY_DIR ${PROJECT_BINARY_DIR})
 set(LLVM_LIT_TOOLS_DIR "${TTMLIR_TOOLCHAIN_DIR}/src/llvm-project/llvm/utils/lit")
+include_directories(${PROJECT_SOURCE_DIR}/include)
```
Comment (nsmithtt) [inline CMakeLists.txt:47]: I think this might be a dup of line 50

**Review by nsmithtt — COMMENTED** [review]
Usman this looks great, thank you, I think it's looking pretty clean.

Can we add it CI?  Enabled on ubuntu, disabled on Mac, that way we kind of get coverage of both build paths?

**Comment by nsmithtt** [comment]
You have some failing tests, because it's disabled in build:

```
FAIL: TTMLIR :: ttmlir/Dialect/TTIR/stablehlo_to_ttir_addop.mlir (24 of 44)
```

See k.in`, as an example see `config.enable_bindings_python`, but we probably want something like:
```python
config.enable_stablehlo = @TTMLIR_ENABLE_STABLEHLO@ and "@TTMLIR_ENABLE_STABLEHLO@" == "ON"
```

And then inside of `test/lit.cfg.py`:
```python
if config.enable_stablehlo:
    config.available_features.add("stablehlo")
```

Finally, inside of the test:
```mlir
// REQUIRES: stablehlo
```

See https://llvm.org/docs/TestingGuide.html for more info.

**Comment by nsmithtt** [comment]
Now failing with:
```
  add_subdirectory given source "/opt/ttmlir-toolchain/src/stablehlo" which
  is not an existing directory.
```

@vmilosevic will the environment build automatically kick off a new docker build?  This used to use GitHub cache

**Comment by vmilosevic** [comment]
> Now failing with:
> 
> ```
>   add_subdirectory given source "/opt/ttmlir-toolchain/src/stablehlo" which
>   is not an existing directory.
> ```
> 
> @vmilosevic will the environment build automatically kick off a new docker build? This used to use GitHub cache
> 
> This commit updates the env

No, it's not automated, you need to manually gick off the docker image build.

image

I added a change to build docker images on self-hosted runner, building large images on Github runners sometimes runs out of space and also fails randomly without any log :(

**Comment by vmilosevic** [comment]
My bad, the docker build was always using code from the main, not the current branch 🤦‍♂️ 
I'm adding a fix now and building the image again.

File: .github/Dockerfile
Comment (nsmithtt) [inline .github/Dockerfile:1]: @vmilosevic , are you good with these changes?

**Review by nsmithtt — APPROVED** [review]
Changes look good on my end, let's get sign off from @vmilosevic on the docker file changes

**Comment by vmilosevic** [comment]
👍 All is good, Docker changes are my commits

**Review by vmilosevic — APPROVED** [review]

**Pull request merged**